src/easydiffraction/io/cif_reader.py

```diff
@@ -1,28 +1,79 @@
 # SPDX-FileCopyrightText: 2024 EasyDiffraction contributors
 # SPDX-License-Identifier: BSD-3-Clause
-# © 2021-2024 Contributors to the EasyDiffraction project <https://github.com/easyscience/EasyDiffraction
+# © 2021-2024 Contributors to the EasyDiffraction project <https://github.com/EasyScience/EasyDiffraction>
 
+import gemmi
 import numpy as np
 
 
+def value_from_loop_by_another_column(block: gemmi.cif.Block,
+                                      category: str,
+                                      desired_column: str,
+                                      another_column: str,
+                                      another_value: str) -> str | None:
+    """
+    Retrieve a value from the desired column in a loop by matching another column's value.
+
+    Parameters:
+        block (gemmi.cif.Block): CIF block.
+        category (str): CIF category name.
+        desired_column (str): Desired column name.
+        another_column (str): Column name to match the value.
+        another_value (str): Value to match in another column.
+
+    Returns:
+        str: The value from the desired column if found, otherwise None.
+    """
+    table = block.find(f'{category}.', [another_column, desired_column])
+    if not table:
+        return None
+
+    another_column_idx = list(table.tags).index(f'{category}.{another_column}')
+    another_column_array = np.array(table.column(another_column_idx))
+    another_value_indices = np.where(another_column_array == another_value)[0]
+
+    if not another_value_indices.size:
+        return None
+
+    desired_column_idx = list(table.tags).index(f'{category}.{desired_column}')
+    desired_column_array = np.array(table.column(desired_column_idx))
+    desired_values = desired_column_array[another_value_indices]
+
+    if not desired_values.size:
+        return None
+
+    return str(desired_values[0])
+
 def pattern_from_cif_block(block) -> dict:
     # Check the experiment type and create the corresponding pattern
     pattern = {}
-    value = block.find_value("_diffrn_radiation_polarization") or block.find_value("_diffrn_radiation.polarization")
+    value = (block.find_value("_diffrn_radiation_polarization")
+             or block.find_value("_diffrn_radiation.polarization"))
```

Comment:
I think we decided to standardize on single quotes throughout. This is a good opportunity to start converting all quoted strings to single quotes.